### ISSUE TITLE
Inconsistent istioctl proxy-status and proxy-config namespace behaviour

### ISSUE THREAD

**Issue by gitfool** [istio/istio#8008]
**Describe the bug**

`istioctl proxy-status` and `istioctl proxy-config` seem to use namespaces differently.

**Expected behavior**

I expect them both to work with global flags (`-n` or `--namespace`) specifying the config namespace.

**Steps to reproduce the bug**

Using `istioctl proxy-status` the namespace needs to be specified as part of the pod name:
```
$ istioctl proxy-status --help

Retrieves last sent and last acknowledged xDS sync from Pilot to each Envoy in the mesh

Usage:
  istioctl proxy-status [<proxy-name>] [flags]

Aliases:
  proxy-status, ps

Examples:
# Retrieve sync status for all Envoys in a mesh
        istioctl proxy-status

# Retrieve sync diff for a single Envoy and Pilot
        istioctl proxy-status istio-egressgateway-59585c5b9c-ndc59.istio-system


Flags:
  -h, --help   help for proxy-status

Global Flags:
      --context string                The name of the kubeconfig context to use
  -i, --istioNamespace string         Istio system namespace (default "istio-system")
  -c, --kubeconfig string             Kubernetes configuration file
      --log_as_json                   Whether to format output as JSON or in plain console-friendly format
      --log_caller string             Comma-separated list of scopes for which to include caller information, scopes can be any of [ads, default, model, rbac]
      --log_output_level string       Comma-separated minimum per-scope logging level of messages to output, in the form of <scope>:<level>,<scope>:<level>,... where scope can be one of [ads, default, model, rbac] and level can be one of [debug, info, warn, error, none] (default "default:info")
      --log_rotate string             The path for the optional rotating log file
      --log_rotate_max_age int        The maximum age in days of a log file beyond which the file is rotated (0 indicates no limit) (default 30)
      --log_rotate_max_backups int    The maximum number of log file backups to keep before older files are deleted (0 indicates no limit) (default 1000)
      --log_rotate_max_size int       The maximum size in megabytes of a log file beyond which the file is rotated (default 104857600)
      --log_stacktrace_level string   Comma-separated minimum per-scope logging level at which stack traces are captured, in the form of <scope>:<level>,<scope:level>,... where scope can be one of [ads, default, model, rbac] and level can be one of [debug, info, warn, error, none] (default "default:none")
      --log_target stringArray        The set of paths where to output the log. This can be any path as well as the special values stdout and stderr (default [stdout])
  -n, --namespace string              Config namespace
  -p, --platform string               Istio host platform (default "kube")
~
$ istioctl proxy-status istio-ingressgateway-6x2xm
Error: unable to parse istio-ingressgateway-6x2xm into a pod name and namespace
~
$ istioctl proxy-status istio-ingressgateway-6x2xm -n istio-system
Error: unable to parse istio-ingressgateway-6x2xm into a pod name and namespace
~
$ istioctl proxy-status istio-ingressgateway-6x2xm.istio-system
--- Pilot Clusters
+++ Envoy Clusters
@@ -201,36 +201,14 @@
             "edsClusterConfig": {
                "edsConfig": {
                   "ads": {

                   }
                },
                "serviceName": "outbound|443||kubernetes.default.svc.cluster.local"
-            },
-            "connectTimeout": "1.000s",
-            "circuitBreakers": {
-               "thresholds": [
-                  {
-
-                  }
-               ]
-            }
-         }
-      },
-      {
-         "cluster": {
-            "name": "outbound|53||kube-dns.kube-system.svc.cluster.local",
-            "type": "EDS",
-            "edsClusterConfig": {
-               "edsConfig": {
-                  "ads": {
-
-                  }
-               },
-               "serviceName": "outbound|53||kube-dns.kube-system.svc.cluster.local"
             },
             "connectTimeout": "1.000s",
             "circuitBreakers": {
                "thresholds": [
                   {

                   }

Listeners Match
Routes Match
```
Using `istioctl proxy-config` the namespace cannot be specified as part of the pod name, but the global flag works:
```
$ istioctl proxy-config --help
A group of commands used to retrieve information about proxy configuration from the Envoy config dump

Usage:
  istioctl proxy-config [command]

Aliases:
  proxy-config, pc

Examples:
  # Retrieve information about proxy configuration from an Envoy instance.
  istioctl proxy-config <clusters|listeners|routes|bootstap> <pod-name>

Available Commands:
  bootstrap   Retrieves bootstrap configuration for the Envoy in the specified pod
  cluster     Retrieves cluster configuration for the Envoy in the specified pod
  listener    Retrieves listener configuration for the Envoy in the specified pod
  route       Retrieves route configuration for the Envoy in the specified pod

Flags:
  -h, --help            help for proxy-config
  -o, --output string   Output format: one of json|short (default "short")

Global Flags:
      --context string                The name of the kubeconfig context to use
  -i, --istioNamespace string         Istio system namespace (default "istio-system")
  -c, --kubeconfig string             Kubernetes configuration file
      --log_as_json                   Whether to format output as JSON or in plain console-friendly format
      --log_caller string             Comma-separated list of scopes for which to include caller information, scopes can be any of [ads, default, model, rbac]
      --log_output_level string       Comma-separated minimum per-scope logging level of messages to output, in the form of <scope>:<level>,<scope>:<level>,... where scope can be one of [ads, default, model, rbac] and level can be one of [debug, info, warn, error, none] (default "default:info")
      --log_rotate string             The path for the optional rotating log file
      --log_rotate_max_age int        The maximum age in days of a log file beyond which the file is rotated (0 indicates no limit) (default 30)
      --log_rotate_max_backups int    The maximum number of log file backups to keep before older files are deleted (0 indicates no limit) (default 1000)
      --log_rotate_max_size int       The maximum size in megabytes of a log file beyond which the file is rotated (default 104857600)
      --log_stacktrace_level string   Comma-separated minimum per-scope logging level at which stack traces are captured, in the form of <scope>:<level>,<scope:level>,... where scope can be one of [ads, default, model, rbac] and level can be one of [debug, info, warn, error, none] (default "default:none")
      --log_target stringArray        The set of paths where to output the log. This can be any path as well as the special values stdout and stderr (default [stdout])
  -n, --namespace string              Config namespace
  -p, --platform string               Istio host platform (default "kube")

Use "istioctl proxy-config [command] --help" for more information about a command.
~
$ istioctl proxy-config listeners istio-ingressgateway-6x2xm
Error: unable to retrieve proxy container name: unable to retrieve Pod: pods "istio-ingressgateway-6x2xm" not found
~
$ istioctl proxy-config listeners istio-ingressgateway-6x2xm -n istio-system
ADDRESS     PORT     TYPE
0.0.0.0     8080     HTTP
0.0.0.0     80       HTTP
~
$ istioctl proxy-config listeners istio-ingressgateway-6x2xm.istio-system
Error: unable to retrieve proxy container name: unable to retrieve Pod: pods "istio-ingressgateway-6x2xm.istio-system" not found
```

**Version**

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.0", GitCommit:"91e7b4fd31fcd3d5f436da26c980becec37ceefe", GitTreeState:"clean", BuildDate:"2018-06-27T20:17:28Z", GoVersion:"go1.10.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.3", GitCommit:"2bba0127d85d5a46ab4b778548be28623b32d0b0", GitTreeState:"clean", BuildDate:"2018-05-28T20:13:43Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
~
$ istioctl version
Version: 1.0.0
GitRevision: 3a136c90ec5e308f236e0d7ebb5c4c5e405217f4
User: root@71a9470ea93c
Hub: gcr.io/istio-release
GolangVersion: go1.10.1
BuildStatus: Clean
```

**Is Istio Auth enabled or not?**
```
$ cat values-overrides.yml
global:
  configValidation: false

ingress:
  enabled: false

gateways:
  istio-ingressgateway:
    type: NodePort
    ports:
    - nodePort: 30080
      port: 80
      name: http-public
    - nodePort: 31080
      port: 8080
      name: http-internal

grafana:
  enabled: true

sidecarInjectorWebhook:
  enabled: false
~
$ sed -i \
    -e 's/kind: Deployment/kind: DaemonSet/' \
    -e 's/replicas:.*//' \
    install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml
~
$ rm -f install/kubernetes/helm/istio/charts/gateways/templates/autoscale.yaml
~
$ helm template install/kubernetes/helm/istio --name istio --namespace istio-system --values values-overrides.yml > ../istio.yml
~
$ kubectl apply -f ../istio.yml

```

**Environment**

AWS EKS

**Cluster state**

.

**Comment by MargarytaChepiga** [comment]
Hello! Is anyone working on this issue?

**Comment by liamawhite** [comment]
Not as far as I know. Would you like to work on it? We should document the proxy name version but support users using either I think.

**Comment by murarisumit** [comment]
@liamawhite if @MargarytaChepiga isn't assigned to it, can I try it out ?

**Comment by liamawhite** [comment]
@murarisumit sure, no one has said they are working on it. I can't assign you but will keep myself assigned. Keep us updated on your progress!

**Comment by liamawhite** [comment]
Resolved by #9533
